### PR TITLE
feat(core): migrate register flow affiliate report logic

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add register flow affiliate report logic.

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/routes/experience/verification-routes/backup-code-verification.ts [hunk 1]
```diff
@@ -52,41 +52,6 @@ export default function backupCodeVerificationRoutes<T extends WithLogContext>(
     }
   );
 
-  router.post(
```
Comment (simeng-li) [inline packages/core/src/routes/experience/verification-routes/backup-code-verification.ts:55]: This is a rebase issue, duplicate code found.

File: packages/core/src/routes/experience/classes/helpers.ts [hunk 2]
```diff
@@ -129,3 +134,31 @@ export const mergeUserMfaVerifications = (
 
   return [...userMfaVerifications, ...newMfaVerifications];
 };
+
+/**
+ * Post affiliate data to the cloud service.
+ *
+ * @remarks forked from {@link routes/interaction/actions/helpers.ts}
```
Comment (gao-sun) [inline packages/core/src/routes/experience/classes/helpers.ts:141]: if we are going to remove the interaction directory, this link would be pointing to the void

File: packages/core/src/routes/experience/classes/libraries/provision-library.ts [hunk 3]
```diff
@@ -262,4 +269,17 @@ export class ProvisionLibrary {
       isInAdminTenant && AdminTenantRole.User,
       isCreatingFirstAdminUser && !isCloud && defaultManagementApiAdminName // OSS uses the legacy Management API user role
     );
+
+  private readonly triggerRegistrationAffiliateAndAnalyticsHooks = ({ id }: User) => {
```
Comment (gao-sun) [inline packages/core/src/routes/experience/classes/libraries/provision-library.ts:273]:
```suggestion
  private readonly triggerAnalyticReports = ({ id }: User) => {
```

File: packages/core/src/routes/experience/classes/helpers.ts (hunk 2 above)
Comment (simeng-li) [reply packages/core/src/routes/experience/classes/helpers.ts]: removed

**Review by gao-sun — APPROVED** [review]

**Pull request merged**